File: PAT/basic_level/1044_火星数字/mars_num.py
from __future__ import print_function

units = ['', 'jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jly', 'aug', 'sep', 'oct', 'nov', 'dec']
tens = ['tret', 'tam', 'hel', 'maa', 'huh', 'tou', 'kes', 'hei', 'elo', 'syy', 'lok', 'mer', 'jou']

def cnv13to10(num):
	l = len(num)
	if l==3:
		for i in range(1, 13):
			if tens[i]==num:
				print(i*13)
				return 0
			elif units[i]==num:
				print(i)
				return 0
			else:
				continue
	elif l==4:
		print(0)
	else:
		sigma = 0
		n1, n2 = num.split(' ')
		for i in range(1, 13):
			if tens[i]==n1:
				sigma = i*13
				break
		for i in range(1, 13):
			if units[i]==n2:
				sigma += i
				break
		print(sigma)
	return 0

def cnv10to13(num):
	num = int(num)
	remain = num % 13
	quotient = num / 13
	if remain==0:
		print(tens[quotient])
	elif quotient==0:
		print(units[remain])
	else:
		print(tens[quotient], units[remain])
	return 0

us_N = raw_input()
for i in range( int(us_N) ):
	num = raw_input()
	if num[0].isdigit():
		cnv10to13(num)
	else:
		cnv13to10(num)


# 10:52 ~ 11:26
# 34min

File: PAT/basic_level/1044_火星数字/mars_num.cpp
#include <iostream>
#include <string>
#include <sstream>
#include <cctype>
using namespace std;

string units[13] = {" ", "jan", "feb", "mar", "apr", "may", "jun", "jly", "aug", "sep", "oct", "nov", "dec"};
string tens[13] = {"tret", "tam", "hel", "maa", "huh", "tou", "kes", "hei", "elo", "syy", "lok", "mer", "jou"};

int str2int(string str){
	int num;
	stringstream ss;
	ss<<str;
	ss>>num;

	return num;
}

int cnv13to10(string str_num){
	int len = str_num.size(), i=0;
	if( len==3 ){
		for( i=1; i<13; i++ ){
			if( tens[i]==str_num ){
				cout<<i*13<<'\n';
				break;
			}else if( units[i]==str_num ){
				cout<<i<<'\n';
				break;
			}else{
				continue;
			}
		}
	}else if( len==4 ){
		cout<<"0\n";
	}else{
		int sigma=0;
		string n1=str_num.substr(0, 3), n2=str_num.substr(4, 3);
		for( i=1; i<13; i++ ){
			if( tens[i]==n1 ){
				sigma = i*13;
				break;
			}
		}
		for( i=1; i<13; i++ ){
			if( units[i]==n2 ){
				sigma += i;
				break;
			}
		}
		cout<<sigma<<'\n';
	}
	return 0;
}

int cnv10to13(string str_num){
	int num = str2int(str_num);
	int remain=num%13, quotient=num/13;
	if( 0==remain )
		cout<<tens[quotient]<<'\n';
	else if( 0==quotient )
		cout<<units[remain]<<'\n';
	else
		cout<<tens[quotient]<<' '<<units[remain]<<'\n';
	return 0;
}

int main(){
	int n;
	string str_num, us_N;

	getline(cin, us_N);
	n = str2int(us_N);
	for( int i=0; i<n; i++ ){
		getline(cin, str_num);
		if( isdigit(str_num[0]) )
			cnv10to13(str_num);
		else
			cnv13to10(str_num);
	}
	return 0;
}

/*
4
29
5
elo nov
tam


*/
